File: src/Commitments.Core/AggregateModel/Behaviour.cs
// Copyright (c) Quinntyne Brown. All Rights Reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

using System.Collections.Generic;


namespace Commitments.Core.AggregateModel;

public class Behaviour: BaseEntity
{
    public int BehaviourId { get; set; }           
    public string Name { get; set; }
    public string Slug { get; set; }
    public string Description { get; set; }
    public int BehaviourTypeId { get; set; }
    public BehaviourType BehaviourType { get; set; }
    public ICollection<Commitment> Commitments { get; set; }
    = new HashSet<Commitment>();
}

File: src/WebApps/Commitments.App/projects/commitments/core/src/lib/model/behaviour/behaviour.ts
// Copyright (c) Quinntyne Brown. All Rights Reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

import { BehaviourType } from "../behaviour-type";

export type Behaviour = {
  behaviourId: string;
  name: string;
  isDesired: boolean;
  description: string;
  behaviourTypeId: number;
  behaviourType: BehaviourType;
};


